### PR TITLE
chore: remove contentType pageComponent property

### PR DESCRIPTION
**Pull request by thiskevinwang**
# Description

Preview: https://dev-portal-3eus23edp-hashicorp.vercel.app/

- remove `pageComponent.contentType`
- remove CurrentContentType contenxt
- remove old search code 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204921002518584

### CODE REVIEW

File: src/types/_app.ts [hunk 1]
```diff
@@ -6,16 +6,12 @@
...
-import { CurrentContentType } from 'contexts'
-import { GlobalThemeOption } from 'styles/themes/types'
 
 type CustomPageComponent<
 	PageProps = Record<string, unknown>,
 	InitialProps = PageProps
 > = NextPage<PageProps, InitialProps> & {
-	contentType?: CurrentContentType
 	layout?: FC
-	theme?: GlobalThemeOption
```
Comment (thiskevinwang) [inline src/types/_app.ts:18]: `theme` removal is some unrelated cleanup

**Review by zchsh — APPROVED** [review]
🙌 This looks great! Search seems to work exactly the same on the deploy preview, and also pulled down locally and tested as well.

🎶 To add a bit of context for posterity, the `contentType` removal definitely qualifies as search-related cruft - my understanding is it was previously used only for the "default search tab selection" before the "All" tab existed 👍 Thanks for this cleanup work 💖 🙇‍♂️
